src/test/java/randomgen/json/RandomJsonWhiteSpaceInjector.java

```diff
@@ -0,0 +1,126 @@
+            whiteSpacesInjectedSoFar++;
+            indexInOriginalByteArray = indexToInsert;
+            indexInResultByteArray = indexToInsert + whiteSpacesInjectedSoFar;
+            indexesToInsertWhiteSpaces.remove(0);
+        }
+        System.arraycopy(
+                originalJsonBytes,
+                indexInOriginalByteArray,
+                resultingJsonBytes,
+                indexInResultByteArray,
+                originalJsonBytes.length - indexInOriginalByteArray);
+        return resultingJsonBytes;
+    }
+
+    private static byte getRandomWhiteSpaceByte() {
+        int randomInt = new Random().nextInt(4);
+        return switch (randomInt) {
+            case 0 -> CARRIAGE_RETURN.getAsciiByteValue();
+            case 1 -> HORIZONTAL_TAB.getAsciiByteValue();
+            case 2 -> LINE_FEED.getAsciiByteValue();
+            case 3 -> SPACE.getAsciiByteValue();
+            default -> throw new IllegalStateException("Unexpected value to get random whitespace byte: " + randomInt);
+        };
+    }
+
+    /**
+     * Checks whether a whitespace can injected before the provided index in the provided JsonBytes.
+     * To keep it simple, this method is sometimes more strict than it needs to be (it returns false
+     * while it was possible, for example in Strings). However, it will never return true if that
+     * would result in invalid JSON if a whitespace would be injected before the provided index.
+     *
+     * @param jsonBytes the JSON bytes for which it will be checked if the whitespace can be
+     *     injected
+     * @param index the index for which it will be checked, must be > 1
+     * @return whether injecting a whitespace before the index in the provided JSON bytes will
+     *     result in valid JSON
+     */
+    private static boolean canInjectWhiteSpaceBeforeByteAtIndex(byte[] jsonBytes, int index) {
+        if (index < 1) {
+            throw new IllegalArgumentException(
+                    "Minimum index to check for is 1 before we won't inject whitespaces before the JSON bytes");
+        }
+        byte originalJsonByte = jsonBytes[index];
+        byte previousOrignalByte = jsonBytes[index - 1];
+        // Not allowed in null values
+        if (isLowercaseU(originalJsonByte) || isLowercaseL(originalJsonByte)) {
+            return false;
+        }
+        // Not allowed in numeric values
+        if (AsciiJsonUtil.isNumericCharacter(originalJsonByte)
+                && AsciiJsonUtil.isNumericCharacter(previousOrignalByte)) {
+            return false;
+        }
+        // Not allowed in boolean values (true/false)
+        if (isLowercaseA(originalJsonByte)
+                || isLowercaseL(originalJsonByte)
+                || isLowercaseS(originalJsonByte)
+                || isLowercaseE(originalJsonByte)
+                || isLowercaseR(originalJsonByte)
+                || isLowercaseU(originalJsonByte)) {
```

Comment:
This is quite cryptic 😅

Reply:
Hahaha I knew it when I wrote it 😅 
The comment on top of it that you cannot add whitespace inside boolean values makes it clear I think :-P

Reply:
Doesn't it miss `t`? Also, it would be slightly better if you'd arrange the order of checks to be something like t, r, u, e, f, a, l, s

Reply:
Or is it just starting from second byte?

Reply:
It is starting from second byte and already in order ;-) (first false, then true), but e is already done in false